src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/VSMSBuildNuGetProjectSystem.cs

```diff
@@ -617,8 +632,11 @@ private static void TrySetCopyLocal(dynamic reference)
                 // In order to properly write this to MSBuild in ALL cases, we have to trigger the Property Change
                 // notification with a new value of "true". However, "true" is the default value, so in order to
                 // cause a notification to fire, we have to set it to false and then back to true
-                reference.CopyLocal = false;
-                reference.CopyLocal = true;
+                if (!reference.CopyLocal)
+                {
+                    reference.CopyLocal = false;
```

Comment:
You can remove setting to false now, since you're only setting it when it's already false and it will cause it to trigger property change event.